pkg/secrets/ansible_vault.go

```diff
@@ -0,0 +1,64 @@
+package secrets
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"syscall"
+
+	vault "github.com/sosedoff/ansible-vault-go"
+	term "golang.org/x/term"
+	yaml "gopkg.in/yaml.v3"
+)
+
+// AnsibleVaultProvider is a provider for ansible-vault files
+type AnsibleVaultProvider struct {
+	vault_file_path string
+	secret          string
+	data            map[interface{}]interface{}
+}
+
+// NewAnsibleVaultProvider creates ad new instance of AnsibleVaultProvider
+func NewAnsibleVaultProvider(vault_path, secret string) (*AnsibleVaultProvider, error) {
+	info, err := os.Stat(vault_path)
+	if err != nil {
+		return nil, fmt.Errorf("error get stat of ansible-vault file: %s", vault_path)
+	}
+	if info.IsDir() {
+		return nil, fmt.Errorf("ansible-vault file is directory: %s", vault_path)
+	}
+	// Decrypt ansible-vault
+	decryptedVault, err := vault.DecryptFile(vault_path, secret)
+	if err != nil {
+		return nil, fmt.Errorf("error decrypting file: %s", vault_path)
+	}
+	log.Printf("[INFO] ansible vault file decrypted")
+
+	// Unmarshal decrypted data
+	m := make(map[interface{}]interface{})
```

Comment:
I don't get this part. is key can be non-string? Same question about value, i.e. why interface{}?

Another question - why the content parsed as yaml? Is it stored as yaml inside the vault? Docs says "You can encrypt single values inside a YAML file using ..." but I think by yaml they mean the playbook. Your code seems to expect key:value yaml inside, and if this is the case we need some clarification with comments explaining how this stored inside with some links to docs.

Reply:
>  Another question - why is the content parsed as yaml?

`YAML` format has been chosen as common for variables in `ansible`.

> Is it stored as yaml inside the vault?

Yes.

> Docs says, "You can encrypt single values inside a YAML file using ..." but I think by yaml they mean the playbook.

Yes, you can use as single encrypted values created with parameter `encrypt-string` [link](https://docs.ansible.com/ansible/latest/vault_guide/vault_encrypting_content.html#creating-encrypted-variables), and this feature has not been implemented. Also you can use full encrypted files [link](https://docs.ansible.com/ansible/latest/vault_guide/vault_encrypting_content.html#encrypting-files-with-ansible-vault).